### ISSUE TITLE
Own implementation of wifi connection

### ISSUE THREAD

**Issue by aemlnt** [HomeSpan/HomeSpan#91]
Hello. I am using my own implementation of the wifi connection. It would be great if homespan could choose which connection method to use. For example: I have third party wifi libraries. They manage the connection. After the home span has verified that the wifi connection is established, it will start working. If not, display a message. In the setting time after the connection is established, it is necessary to establish the connection separately for homespan.

**Comment by HomeSpan** [comment]
This is a reasonable idea but can you provide some more details on how you would want to integrate a separate wifi library into the existing code?  Note that the WiFi stack is not just used to make the initial connection - it is also used continuously during runtime to support the server, manage all incoming traffic, start and stop connections, and provide all the MDNS broadcasts needed for HomeKit.  All of that would need to be replaced in HomeSpan with different code.

**Comment by aemlnt** [comment]
I have an idea how to make it very simple. We could just pass the wifi login parameters (ssid and password) to homespan.

Example: the wifi manager library allows you to get the saved network data via the "getWiFiPass" and "getWiFiSSID" variables. We could just use the received data to substitute it into homespan.

And no need to rewrite the whole stack)

**Comment by HomeSpan** [comment]
Using an external library just for the initial identification of the SSID and WiFi Password (as opposed to initiating the actual connection each time) should doable.  The easiest way would be if you saved the SSID/Pwd you acquire in the NVS where HomeSpan normally would place it.  It would then get picked up during startup, could be erased using the normal HomeSpan commands, etc.

Two steps would be needed to make this work.  The first is a method enabling you to call your library so it could do whatever it needs to do to acquire an SSID/Pwd.  Based on another user's request, I've already added a new HomeSpan feature that allows you to extend the HomeSpan CLI with your own commands.  For example, you can add your own version of the 'W' command that would call your library instead of the current code that asks the user for an SSID/Pwd.  This new method, called `SpanUserCommand`, has been added to the dev branch so you can try it out now if you'd like (it will be part of the next patch update).

The second step would be to then save the SSID/Pwd you acquired in the NVS.  I could create a method for this, or you can simply use the ESP32's NVS functionality directly.  To store WiFi Credentials in the NVS, use the following:

```C++
nvs_set_blob(HAPClient::wifiNVS,"WIFIDATA",&network.wifiData,sizeof(network.wifiData));
nvs_commit(HAPClient::wifiNVS);
```

where the structure `wifiData` contains your SSID/Pwd as follows:

```C++
struct {
  char ssid[33];
  char pwd[65];
} wifiData;
```

**Comment by HomeSpan** [comment]
These features have now been implemented in version 1.3.0.  See `void setApFunction(void (*func)())` and `void setWifiCredentials(const char *ssid, const char *pwd)` in [API Reference](https://github.com/HomeSpan/HomeSpan/blob/master/docs/Reference.md) for details.
